erpnext/stock/doctype/material_request/material_request.py

```diff
@@ -329,3 +334,36 @@ def set_missing_values(source, target):
 	}, target_doc, set_missing_values)
 
 	return doclist
+
+@frappe.whitelist()
+def raise_production_orders(source_name):
+	material_request= frappe.get_doc("Material Request", source_name)
+	errors =[]
+	production_orders = []
+	for d in material_request.items:
+		if (d.qty - d.ordered_qty) >0 :
+			if frappe.db.get_value("Item", d.item_code, "is_pro_applicable"):
+				prod_order = frappe.new_doc("Production Order")
+				prod_order.production_item = d.item_code
+				prod_order.qty = d.qty - d.ordered_qty
+				prod_order.fg_warehouse = d.warehouse
+				prod_order.description = d.description
+				prod_order.stock_uom = d.uom
+				prod_order.expected_delivery_date = d.schedule_date
+				prod_order.sales_order = d.sales_order
+				prod_order.bom_no = get_item_details(d.item_code).bom_no
+				prod_order.material_request = material_request.name
+				prod_order.material_request_item = d.name
+				prod_order.planned_start_date = material_request.transaction_date
+				prod_order.company = material_request.company
+				prod_order.save()
+				production_orders.append(prod_order.name)
+			else:
+				errors.append(d.item_code + " in Row " + cstr(d.idx))
+	if production_orders: 
+		message = ["""<a href="#Form/Production Order/%s" target="_blank">%s</a>""" % \
+			(p, p) for p in production_orders]
+		msgprint(_("Production Orders {0} created").format(comma_and(message)))
+	if errors:
+		msgprint(_("Could not Raise Production Orders for {0}").format(comma_and(errors)))
```

Comment:
Instead of comma separated string, use newline separated for such cases for better readibility.

Also state why it is not created.